repondeur/zam_repondeur/resources.py

```diff
@@ -0,0 +1,179 @@
+from typing import Iterator, List, Optional
+
+from pyramid.request import Request
+
+from zam_repondeur.models import (
+    Amendement as AmendementModel,
+    Lecture as LectureModel,
+    DBSession,
+)
+
+
+class Resource(dict):
+    """Location-aware resource"""
+
+    __name__: Optional[str] = None
+    __parent__: Optional["Resource"] = None
```

Comment:
Why `__parent__` over `parent`? We use it in templates so it doesn't look private nor magic to me.

Reply:
That's the protocol used by Pyramid: https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/resources.html#location-aware-resources

I think they used reserved attribute names so that it will not clash with existing attributes you might have on your resources.

Reply:
Anyway, I'll add a `parent` property so that we don't have to access `__parent__` directly, and to make overriding the type in subclasses easier.